### PR TITLE
[CDEC-466] Remove partial field accessors from `PollVerFailure` datatype

### PR DESCRIPTION
**Pull request by Jimbo4350**
## Description

Remove partial field accessors from `PollVerFailure` datatype.

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CDEC-466

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: update/src/Pos/Update/Logic/Local.hs [hunk 1]
```diff
@@ -79,7 +79,7 @@ getMemPool
     :: (MonadReader ctx m, HasLens UpdateContext ctx UpdateContext, MonadIO m)
     => m MemPool
 getMemPool = msPool <$>
-    (atomically . readTVar . mvState =<< views (lensOf @UpdateContext) ucMemState)
+    (readTVarIO . mvState =<< views (lensOf @UpdateContext) ucMemState)
```
Comment (mhuesch) [inline update/src/Pos/Update/Logic/Local.hs]: 👍 this is cleaner, and seems OK to update in this PR, even though it's unrelated to partial field accessors.

File: update/src/Pos/Update/Poll/Failure.hs [hunk 2]
```diff
@@ -22,87 +22,121 @@ import           Pos.Crypto (shortHashF)
...
+    --      ptopUpId
+    | PollMoreThanOneProposalPerEpoch !StakeholderId !UpId
+    -- | PollUnknownAttributesInProposal
+    --       puapUpId
+    --       puapAttrs
+    | PollUnknownAttributesInProposal !UpId !UpAttributes
+    -- | PollTipMismatch
+    --       ptmTipDB
+    --       ptmTipMemory
+    | PollTipMismatch !HeaderHash !HeaderHash
```
Comment (mhuesch) [inline update/src/Pos/Update/Poll/Failure.hs]: 👍 I'm liking this new style with the comment-ed field names above the constructs.

File: update/src/Pos/Update/Logic/Local.hs (hunk 1 above)
Comment (Jimbo4350) [reply update/src/Pos/Update/Logic/Local.hs]: We can thank @erikd 's pre-commit hook for this :)

Comment (mhuesch) [reply update/src/Pos/Update/Logic/Local.hs]: Ah right, I forgot about HLint.

File: update/src/Pos/Update/Poll/Failure.hs [hunk 3]
```diff
@@ -22,87 +22,121 @@ import           Pos.Crypto (shortHashF)
...
+    | PollProposalIsDecided !UpId !StakeholderId
+    -- | PollExtraRevote
+    --       perUpId
+    --       perStakeholder
+    --       perDecision
+    | PollExtraRevote !UpId !StakeholderId !Bool
+    -- | PollWrongHeaderBlockVersion
+    --       pwhpvGiven
+    --       pwhpvAdopted
+    | PollWrongHeaderBlockVersion !BlockVersion !BlockVersion
```
Comment (intricate) [inline update/src/Pos/Update/Poll/Failure.hs]: Should we maybe use some `newtype`s for some of these constructor fields?

Comment (intricate) [reply update/src/Pos/Update/Poll/Failure.hs]: or am I going overboard?

Comment (erikd) [reply update/src/Pos/Update/Poll/Failure.hs]: This is an error constructor. The absolute worst thing that can happen here is that the error message is wrong. In this casned, I'm ok with leaving it as it is :). If this was something like a transaction with source and destination addresses, then a `newtype` would make a **lot** of sense.

Comment (mhuesch) [reply update/src/Pos/Update/Poll/Failure.hs]: Agree with @erikd that it doesn’t seem critical in this case.

Comment (intricate) [reply update/src/Pos/Update/Poll/Failure.hs]: @erikd @mhuesch Thanks guys

**Pull request merged**